specs/pvs/corla.pvs

```diff
@@ -1650,6 +1772,186 @@ END system_assumptions
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+%%% \section{System Logging}
+
+%%% The system must log a variety of events. One reason to log
+%%% information is to help understand how the system is operating, fix
+%%% bugs post-facto, understand how users are using the system,
+%%% etc. Another, more important reason to log in this context of this
+%%% system is to provide an indeliable record of administrator and
+%%% auditor actions so that any audit can be "replayed" by any third
+%%% party.
+
+logging: THEORY
+BEGIN
+  log: TYPE = sequence[string]
+  chain_of_custody_log: TYPE FROM log
+% @trace Per email from Jerome Lovato at CDOS on 19 July 2017: Logging
+% of Audit Board and SOS user errors during use of the system [...is
+% the only logging we care about].
+  rla_tool_log: TYPE FROM log
+END logging
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%%% \section{Data Model}
```

Comment:
And here's the Django code for batch-level comparison auditing from electionaudits, used to run the Boulder RLA in 2010, and the more extensive almost-RLA in 2008. 
 Classes for CountyElection, Contest, Batch, ContestBatch, Choice, VoteCount, Margin http://bazaar.launchpad.net/~nealmcb/electionaudits/trunk/view/head:/electionaudits/models.py